client/internal/config.go

```diff
@@ -203,6 +215,17 @@ func ReadConfig(input ConfigInput) (*Config, error) {
 		config.WgPort = iface.DefaultWgPort
 		refresh = true
 	}
+
+	if input.WgIface == "" {
+		config.WgIface = iface.WgInterfaceDefault
+		refresh = true
+	}
+
+	if !slices.Contains(config.IFaceBlackList, config.WgIface) {
```

Comment:
You can probably store the previous WgIface and if changed, loop through the config.IFaceBlackList and replace the matching element index with the new interface.

Reply:
Sorry I wasn't clear on my idea. I've left a code suggestion.